kv/memberlist/memberlist_client_test.go

```diff
@@ -519,59 +526,163 @@ func TestMultipleCAS(t *testing.T) {
 	}
 }
 
+func defaultKVConfig(i int) KVConfig {
+	id := fmt.Sprintf("Member-%d", i)
+	var cfg KVConfig
+	flagext.DefaultValues(&cfg)
+	cfg.NodeName = id
+
+	cfg.GossipInterval = 100 * time.Millisecond
+	cfg.GossipNodes = 3
+	cfg.PushPullInterval = 5 * time.Second
+
+	cfg.TCPTransport = TCPTransportConfig{
+		BindAddrs: []string{"localhost"},
+		BindPort:  0, // randomize ports
+	}
+
+	return cfg
+}
+
 func TestMultipleClients(t *testing.T) {
-	c := dataCodec{}
+	members := 10
 
-	const members = 10
-	const key = "ring"
+	err := testMultipleClientsWithConfigGenerator(t, members, defaultKVConfig)
+	require.NoError(t, err)
+}
 
-	var clients []*Client
+func TestMultipleClientsWithMixedLabelsAndExpectFailure(t *testing.T) {
+	members := 3
+	// We want 3 members, they will be configured with the following labels:
+	// 1) ""
+	// 2) "label1"
+	// 3) "label2"
+	//
+	// We expect that it won't be possible to build a memberlist cluster with mixed labels.
+	var labels = []string{
+		"",
+		"label1",
+		"label2",
+	}
 
-	stop := make(chan struct{})
-	start := make(chan struct{})
+	configGen := func(i int) KVConfig {
+		cfg := defaultKVConfig(i)
 
-	port := 0
+		cfg.SkipInboundLabelCheck = false
+		cfg.Label = labels[i]
 
-	for i := 0; i < members; i++ {
-		id := fmt.Sprintf("Member-%d", i)
-		var cfg KVConfig
-		flagext.DefaultValues(&cfg)
-		cfg.NodeName = id
+		return cfg
+	}
 
-		cfg.GossipInterval = 100 * time.Millisecond
-		cfg.GossipNodes = 3
-		cfg.PushPullInterval = 5 * time.Second
+	err := testMultipleClientsWithConfigGenerator(t, members, configGen)
```

Comment:
[nit] We can remove `members` variable at all.

```suggestion
	err := testMultipleClientsWithConfigGenerator(t, len(labels), configGen)
```